user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -418,66 +421,15 @@ private CompletionStage<Void> reconcileCredentialsQuotasAndAcls(Reconciliation r
      * @return                  CompletionStage describing the result
      */
     private CompletionStage<ReconcileResult<Secret>> reconcileUserSecret(Reconciliation reconciliation, KafkaUserModel user, Secret currentSecret, KafkaUserStatus userStatus) {
-        return CompletableFuture.supplyAsync(() -> {
-            String namespace = reconciliation.namespace();
-            String name = user.getSecretName();
-            Secret desiredSecret = user.generateSecret();
-
-            if (desiredSecret != null)  {
-                if (currentSecret != null)  {
-                    // Both secrets exist => if they differ we patch them, if not we just continue
-                    if (new ResourceDiff<>(reconciliation, "Secret", name, currentSecret, desiredSecret, ResourceDiff.DEFAULT_IGNORABLE_PATHS).isEmpty()) {
-                        // Secrets are identical
-                        LOGGER.debugCr(reconciliation, "Secret {}/{} exist, and is identical", namespace, name);
-                        userStatus.setSecret(desiredSecret.getMetadata().getName());
-                        return ReconcileResult.noop(desiredSecret);
-                    } else {
-                        // Secrets differ
-                        LOGGER.debugCr(reconciliation, "Secret {}/{} exist, patching it", namespace, name);
-                        client.secrets().inNamespace(namespace).resource(desiredSecret).update();
-                        userStatus.setSecret(desiredSecret.getMetadata().getName());
-                        return ReconcileResult.patched(desiredSecret);
-                    }
-                } else {
-                    LOGGER.debugCr(reconciliation, "Secret {}/{} does not exist, creating it", namespace, name);
-                    createOrReplaceSecret(reconciliation, namespace, desiredSecret);
-                    userStatus.setSecret(desiredSecret.getMetadata().getName());
-                    return ReconcileResult.created(desiredSecret);
+        return secretOperator
+            .reconcile(reconciliation, reconciliation.namespace(), user.getSecretName(), currentSecret, user.generateSecret())
+            .whenComplete((result, error) -> {
+                if (error == null) {
```

Comment:
Yes, it would be propagated. The `CompletionStage` returned by the method would still be in a failed state. In other words, `whenComplete` doesn't change the result of the stage it is called on.